lib/mapObjects/CGTownInstance.cpp

```diff
@@ -757,6 +757,9 @@ bool CGTownInstance::allowsTrade(EMarketMode mode) const
 
 	case EMarketMode::RESOURCE_SKILL:
 		return hasBuilt(BuildingSubID::MAGIC_UNIVERSITY);
+	case EMarketMode::CREATURE_EXP:
+	case EMarketMode::ARTIFACT_EXP:
+		return false;
```

Comment:
I'm not sure how to correctly connect the market to the grotto. So for now it's always false.